drgn_tools/iscsi.py

```diff
@@ -77,28 +82,34 @@ def for_each_iscsi_session(prog: Program) -> Iterator[Object]:
     :returns: a iterator of ``struct iscsi_session *``
 
     """
-    for h in for_each_iscsi_host(prog):
-        yield __get_iscsi_session(h)
+    for h in for_each_scsi_host(prog):
+        yield __get_iscsi_session(prog, h)
```

Comment:
The system could contain other scsi disk like virtio-scsi, sas or fc disk other than iscsi, here it is iterating scsi host, it will include all other types as well, we should provide a new helper `for_each_iscsi_host` which used `for_each_scsi_host` and then just return the host which is iscsi type.